src/common_types.hpp

```diff
@@ -21,6 +21,10 @@ using cell_gid_type = std::uint32_t;
 
 using cell_size_type = typename std::make_unsigned<cell_gid_type>::type;
 
+// For cluster domain
```

Comment:
Right -- needs more words to explain that this is an id for domains in the sense of node number in a cluster, aka some source that isn't local to the shared memory.